x/concentrated-liquidity/model/codec.go

```diff
@@ -5,14 +5,22 @@ import (
 	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/msgservice"
+
+	swaproutertypes "github.com/osmosis-labs/osmosis/v13/x/swaprouter/types"
 )
 
 func RegisterCodec(cdc *codec.LegacyAmino) {
-	cdc.RegisterConcrete(&Pool{}, "osmosis/concentratedliquidity/ConcentratedLiquidityPool", nil)
-	cdc.RegisterConcrete(&MsgCreateConcentratedPool{}, "osmosis/concentratedliquidity/create-concentrated-pool", nil)
+	cdc.RegisterConcrete(&Pool{}, "osmosis/cl-pool", nil)
+	cdc.RegisterConcrete(&MsgCreateConcentratedPool{}, "osmosis/cl-create-pool", nil)
 }
 
 func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
+	registry.RegisterInterface(
+		"osmosis.swaprouter.v1beta1.PoolI",
+		(*swaproutertypes.PoolI)(nil),
+		&Pool{},
+	)
```

Comment:
Why do we marshal an interface though? I think this could have been avoided if we marshaled a pool struct directly

Reply:
The reason we marshal an interface in gamm is that there are 2 implementations - balancer and stableswap. Here, we only have one pool model. So we should be able to marshal its concrete type directly

Reply:
@p0mvn wdym by marshalling a pool directly?

Reply:
I'm pretty sure he means not marshaling into proto any and instead directly marshaling into the struct itself, but sometimes this isnt possible due to circular dependencies in the proto files

Reply:
Hrmm, it potentially makes sense to register this, if we wanted a swaprouter query for underlying structs as an interface tho

Reply:
But I do agree, that serialization into state should just be the concrete struct, and such a register would just be for a one-day hypothetical query

Reply:
FWIW, I don't really care if we serialize into state as concrete struct vs interface.

Concrete is slightly more space efficient, but I doubt this efficiency ever matters. Interface could plausibly give more ease of switching in the future, but if we didn't do an interface I don't think it'd be that hard to change in the future either.